.github/container/Dockerfile.triton

```diff
@@ -17,48 +18,62 @@ ENV TRITON_NVDISASM_PATH=/usr/local/cuda/bin/nvdisasm
-# patches are available under patches/openxla-triton
-RUN git-clone.sh ${URLREF_OXLA_TRITON} ${SRC_PATH_OXLA_TRITON}
+# Use XLA's Bazel configuration to get the relevant tag from the openxla/triton
+# fork's llvm-head branch and apply XLA's extra patches to it
+FROM ${BASE_IMAGE} as checkout
+ARG SRC_PATH_JAX
+ARG SRC_PATH_XLA
 RUN <<"EOF" bash -ex
-shopt -s nullglob
-cd /opt/openxla-triton
-for patch in /opt/manifest.d/patches/openxla-triton/*.patch; do
-  patch -p1 < "${patch}"
-done
-git diff
+pushd "${SRC_PATH_XLA}"
+BAZEL=$(find "${SRC_PATH_JAX}/build" -type f -executable -name 'bazel-*')
+"${BAZEL}" --output_base=/opt/triton-checkout fetch @triton//:BUILD
 EOF
 
+###############################################################################
+## Get (only) the Triton source from the checkout step
+###############################################################################
+FROM base as builder
+ARG SRC_PATH_TRITON
+ARG SRC_PATH_XLA
+COPY --from=checkout /opt/triton-checkout/external/triton "${SRC_PATH_TRITON}"
 RUN <<"EOF" bash -ex
-mkdir -p "${SRC_PATH_OXLA_TRITON}/dist"
-# This sidesteps the RTTI-related error in https://github.com/openai/triton/pull/3213 that otherwise breaks container
-# builds with openxla/triton tag cl608559313
-sed -i 's|^add_subdirectory(unittest)|# unit tests disabled|' "${SRC_PATH_OXLA_TRITON}/CMakeLists.txt"
-sed -i 's|backends = _copy_backends(\["nvidia", "amd"\])|backends = _copy_backends(["nvidia"])|g' "${SRC_PATH_OXLA_TRITON}/python/setup.py"
-sed -i '1s|^|include_directories(${CMAKE_SOURCE_DIR}/third_party/nvidia/backend/include)\n|' "${SRC_PATH_OXLA_TRITON}/lib/Conversion/TritonGPUToLLVM/CMakeLists.txt"
-pip wheel --wheel-dir="${SRC_PATH_OXLA_TRITON}/dist" "${SRC_PATH_OXLA_TRITON}/python"
+mkdir -p "${SRC_PATH_TRITON}/dist"
+sed -i -e 's|^add_subdirectory(unittest)|# unit tests disabled|' "${SRC_PATH_TRITON}/CMakeLists.txt"
+sed -i -e 's|BackendInstaller.copy(\["nvidia", "amd"\])|BackendInstaller.copy(["nvidia"])|g' "${SRC_PATH_TRITON}/python/setup.py"
+# Extra patches to Triton maintained in XLA. These are already applied in the checkout stage.
+XLA_TRITON_PATCHES="${SRC_PATH_XLA}/third_party/triton"
+# This patch only works with a newer LLVM than is pinned in the branch.
+PATCH_FILE="${XLA_TRITON_PATCHES}/llvm_integration/cl623185214.patch"
+if [[ $(cat "${SRC_PATH_TRITON}/cmake/llvm-hash.txt") == "6f44bb7717897191be25aa01161831c67cdf5b84" && -f "${PATCH_FILE}" ]]; then
+  pushd "${SRC_PATH_TRITON}"
+  patch -p1 -R < "${PATCH_FILE}"
+  popd
+fi
+# This patch adds two files that are not known to CMake
+if [[ -f "${XLA_TRITON_PATCHES}/xla_extensions/sparse_dot_base.patch" ]]; then
+  sed -i -e '/ConvertLayoutOpToLLVM.cpp/a ConvertLayoutOpToLLVM/SharedToSparseDotOperand.cpp' "${SRC_PATH_TRITON}/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/CMakeLists.txt"
+  sed -i -e '/DotOpToLLVM.cpp/a DotOpToLLVM/Sparse.cpp' "${SRC_PATH_TRITON}/third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/CMakeLists.txt"
+fi
```

Comment:
Maybe it is time for a rosetta-triton build? Adding patches to an upstream diverges a bit from our best practice. Unless I'm missing some context here which is definitely not unlikely (this is not to block this PR but as a note for ourselves, we can tackle this later too)

Reply:
This Dockerfile is trying to replicate the patched tag of openxla/triton described by this file: https://github.com/openxla/xla/blob/2aa2b0fec91c77805928ff368d1711e74697d171/third_party/triton/workspace.bzl#L11-L19.
The inconsistency that `sed` is trying to bridge here is that that version is intended to be built with Bazel, but in this Dockerfile we are trying to build a Python wheel using CMake: when patches add entirely new files, CMake is unaware of them and needs to be told.